tests/src/test/java/alluxio/client/fs/FileOutStreamAsyncWriteIntegrationTest.java

```diff
@@ -24,40 +24,88 @@
+import alluxio.testutils.BaseIntegrationTest;
 import alluxio.testutils.IntegrationTestUtils;
 import alluxio.testutils.LocalAlluxioClusterResource;
 import alluxio.util.CommonUtils;
 import alluxio.util.FormatUtils;
 import alluxio.util.io.PathUtils;
 import alluxio.wire.FileBlockInfo;
-import alluxio.wire.WorkerInfo;
-import alluxio.worker.block.BlockMasterSync;
 import alluxio.worker.block.BlockWorker;
 import alluxio.worker.block.SpaceReserver;
 
+import org.junit.After;
+import org.junit.AfterClass;
 import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TestRule;
 import org.powermock.reflect.Whitebox;
 
-import java.util.Arrays;
-
 /**
  * Integration tests for {@link alluxio.client.file.FileOutStream} of under storage type being async
  * persist.
  */
-public final class FileOutStreamAsyncWriteIntegrationTest
-    extends AbstractFileOutStreamIntegrationTest {
-
+public final class FileOutStreamAsyncWriteIntegrationTest extends BaseIntegrationTest {
   private static final String TINY_WORKER_MEM = "512k";
   private static final String TINY_BLOCK_SIZE = "16k";
 
+  protected static LocalAlluxioJobCluster sLocalAlluxioJobCluster;
+
+  @ClassRule
+  public static LocalAlluxioClusterResource sLocalAlluxioClusterResource =
+      buildLocalAlluxioClusterResource();
+
+  public static LocalAlluxioClusterResource buildLocalAlluxioClusterResource() {
+    LocalAlluxioClusterResource.Builder resource = new LocalAlluxioClusterResource.Builder()
+        .setProperty(PropertyKey.USER_FILE_BUFFER_BYTES, "8k")
+        .setProperty(PropertyKey.USER_FILE_REPLICATION_DURABLE, 1)
+        .setProperty(PropertyKey.MASTER_PERSISTENCE_SCHEDULER_INTERVAL_MS, "100ms")
+        .setProperty(PropertyKey.MASTER_PERSISTENCE_CHECKER_INTERVAL_MS, "100ms")
+        .setProperty(PropertyKey.WORKER_TIERED_STORE_RESERVER_INTERVAL_MS, "100ms")
+        .setProperty(PropertyKey.WORKER_MEMORY_SIZE, TINY_WORKER_MEM)
+        .setProperty(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT, TINY_BLOCK_SIZE);
+    return resource.build();
+  }
+
+  @Rule
+  public TestRule mResetRule = sLocalAlluxioClusterResource.getResetResource();
+
+  private FileSystem mFileSystem;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    sLocalAlluxioJobCluster = new LocalAlluxioJobCluster();
+    sLocalAlluxioJobCluster.setProperty(PropertyKey.JOB_MASTER_WORKER_HEARTBEAT_INTERVAL, "25ms");
+    sLocalAlluxioJobCluster.start();
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    if (sLocalAlluxioJobCluster != null) {
+      sLocalAlluxioJobCluster.stop();
+    }
+  }
+
+  @Before
+  public void before() throws Exception {
+    mFileSystem = sLocalAlluxioClusterResource.get().getClient();
+  }
+
+  @After
+  public void after() throws Exception {
+    mFileSystem.close();
```

Comment:
are we supposed to close the clients we get from the local alluxio cluster?

Reply:
yes, otherwise it will result in resource leaks